src/main/java/com/jvavateam/carsharingapp/service/impl/RentalServiceImpl.java

```diff
@@ -75,9 +77,16 @@ public List<RentalResponseDto> getAllByManager(RentalSearchParameters searchPara
                                                    Pageable pageable) {
         Specification<Rental> searchSpecification =
                 rentalSpecificationBuilder.build(searchParameters);
-        return rentalRepository.findAll(searchSpecification, pageable).stream()
-                .map(rentalMapper::toDto)
-                .toList();
+        try {
+            return rentalRepository.findAll(searchSpecification, pageable).stream()
+                    .map(rentalMapper::toDto)
+                    .toList();
+        } catch (NoSuchElementException e) {
+            throw new InvalidRequestParametersException("Entered wrong search params. "
+                    + "Entered: \nUser id: " + searchParameters.userId()
+                    + "\nIs active: " + searchParameters.isActive()
+            );
```

Comment:
It doesn't fix the issue. The exception occurs when you try to build a Specification.
Look at the attached picture and replace NoSuchElementException with the corresponding Custom Exception (e.g. NoSuchSpecificationProviderExceprion)

![image](https://github.com/jv-jun23-jvavaTeam/car-sharing-app/assets/113283324/7d7955c2-3972-45cc-8dc5-90e3af922975)